dotCMS/src/main/java/com/dotcms/contenttype/business/StoryBlockAPIImpl.java

```diff
@@ -364,10 +355,50 @@ private void refreshBlockEditorDataMap(final Map<String, Object> dataMap, final
     private void addContentletRelationships(final Contentlet contentlet) {
         final HttpServletRequest httpRequest = HttpServletRequestThreadLocal.INSTANCE.getRequest();
         final PageMode currentPageMode = PageMode.get(httpRequest);
-        if (null != httpRequest && null == httpRequest.getAttribute(WebKeys.HTMLPAGE_DEPTH)) {
-            httpRequest.setAttribute(WebKeys.HTMLPAGE_DEPTH, MAX_RELATIONSHIP_DEPTH.get());
+
+        int depth = getInitialDepthValue();
+
+        if (isInsideAnotherBlockEditorAndRelatedContent()) {
+            depth = decreaseDepthValue(depth);
+        }
+
+        ContentUtils.addRelationships(contentlet, APILocator.systemUser(), currentPageMode, contentlet.getLanguageId(), depth);
+    }
+
+    private int decreaseDepthValue(int depthValue) {
```

Comment:
Let's add as much Javadoc as possible to provide as much info to other devs as possible